src/crimson/net/SocketMessenger.cc

```diff
@@ -61,8 +61,18 @@ seastar::future<> SocketMessenger::bind(const entity_addrvec_t& addrs)
   }
   logger().info("listening on {}", my_addrs.front().in4_addr());
   return container().invoke_on_all([my_addrs](auto& msgr) {
-      msgr.do_bind(my_addrs);
-    });
+    msgr.do_bind(my_addrs);
+  }).handle_exception_type([this] (const std::system_error& e) {
```

Comment:
why do we need to capture `this`?

Reply:
logger needs `this` to print out messenger info as prefix.